python-lib/gluonts_forecasts/model_handler.py

```diff
@@ -127,71 +40,57 @@ class ModelParameterError(ValueError):
 class ModelHandler:
```

Comment:
I would call it Model, but then we have a conflict with another class named Model too. What does ModelHandler do differently compared to Model?
Perhaps there is an easier way to structure all of this?

Reply:
ModelHandler is used to define some gluon models properties and calls directly Gluon methods (but also register some non-gluon models) and Model is our abstraction on a model used to train and evaluate

Reply:
I still struggle with understanding the names. Instinctively, when I read the code, I would want to use Model directly, because it has a straightforward name.
So Model and ModelHandler are two different abstractions on top of GluonTS. But what is the distinct goal of each abstraction? Is there any way we could have a single abstraction? Or change the term "handle" to something more specific?

Reply:
ModelHandler is really only about defining some model specifications, it has no methods that do something, whereas Model is a class that does stuff like training or evaluation, maybe ModelHandler could be named ModelSpecificator or ModelConfigurator ?

Reply:
How about just ModelConfig? The "..or" formulation sounds like the class is going to change the Model. If I understand correctly, it's more reading info about the model than changing it.

Reply:
yes sounds good !! so to summarize, ModelConfig, ModelConfigRegistry and in the Model class self.model_config ?

Reply:
tip top

Reply:
b759999